alpine-glibc/Dockerfile

```diff
@@ -0,0 +1,49 @@
+ARG ARCH=""
+FROM ${ARCH}debian:buster-slim
+
+RUN \
+    apt-get update && apt-get install -y --no-install-recommends \
+      ca-certificates \
+      gcc \
+      netbase \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN set -x \
+    && mkdir -p rootfs/lib \
+    && set -- \
+        /etc/nsswitch.conf \
+        /etc/ssl/certs/ca-certificates.crt \
+        /usr/share/zoneinfo \
+        /etc/services \
+        /lib/"$(gcc -print-multiarch)"/libpthread.so.* \
+    && while [ "$#" -gt 0  ]; do \
+        f="$1"; shift; \
+        fn="$(basename "$f")"; \
+        if [ -e "rootfs/lib/$fn" ]; then continue; fi; \
+        if [ "${f#/lib/}" != "$f" ]; then \
+            ln -vL "$f" "rootfs/lib/$fn"; \
+        else \
+            d="$(dirname $f)" \
+            && mkdir -p "rootfs/${d#/}" \
+            && cp -av "$f" "rootfs/${f#/}"; \
+        fi; \
+    done
+
+FROM        ${ARCH}alpine:latest as alpine
+RUN apk add busybox-static && \
+    rm /bin/* && \
+    /tmp/busybox --install /bin && \
+    mv /tmp/busybox /bin/
+
+FROM        ${ARCH}busybox:glibc
```

Comment:
Why not `FROM scratch`? If we're not going to use the busybox binary from this package, why use this as the image? All of the prometheus project binaries are static, we don't actually use glibc anymore.

Also, maybe we can simplify this to just pull the misc etc files from alpine as well, skip the debian layer as well.

Reply:
I consume this image: https://hub.docker.com/r/prom/prometheus

I opened this PR to try to produce a more secure base image for that image to be based on so I (and every other consumer) don't have to repackage it for `busybox` security issues.  I was trying to do this with the least set of changes in order to have the least risk of breakages and to respect the intent of the existing images. 

As I noted above, I believe:
> From the perspective of consumers of these images, the key features are:
> * The busybox toolset
> * The specified libc flavor

This PR provides the above with the least changes from the existing behavior.  

What you're suggesting is I skip this and go make the case for prometheus `scratch` images. Fair enough.

Reply:
Yeah I don't get why this isn't `FROM scratch` either. I would also assume that the final image will contain both the original busybox (in a hidden layer) as well as the new one.